src/backend/metal/src/conversions.rs

```diff
@@ -1,114 +1,118 @@
-        //f::Bc7Unorm          => BC7_RGBAUnorm,
-        //f::Bc7Srgb           => BC7_RGBAUnorm_sRGB,
-        //f::EacR11Unorm       => EAC_R11Unorm, // !macOS
-        //f::EacR11Inorm       => EAC_R11Snorm, // !macOS
-        //f::EacR11g11Unorm    => EAC_RG11Unorm, // !macOS
-        //f::EacR11g11Inorm    => EAC_RG11Snorm, // !macOS
-        //f::Etc2R8g8b8Unorm   => ETC2_RGB8, // !macOS
-        //f::Etc2R8g8b8Srgb    => ETC2_RGB8_sRGB, // !macOS
-        //f::Etc2R8g8b8a1Unorm => ETC2_RGB8A1, // !macOS
-        //f::Etc2R8g8b8a1Srgb  => ETC2_RGBA1_sRGB, // !macOS
-        //f::Astc4x4Unorm      => ASTC_4x4_LDR, // !macOS
-        //f::Astc4x4Srgb       => ASTC_4x4_sRGB, // !macOS
-        //f::Astc5x4Unorm      => ASTC_5x4_LDR, // !macOS
-        //f::Astc5x4Srgb       => ASTC_5x4_sRGB, // !macOS
-        //f::Astc5x5Unorm      => ASTC_5x5_LDR, // !macOS
-        //f::Astc5x5Srgb       => ASTC_5x5_sRGB, // !macOS
-        //f::Astc6x5Unorm      => ASTC_6x5_LDR, // !macOS
-        //f::Astc6x5Srgb       => ASTC_6x5_sRGB, // !macOS
-        //f::Astc6x6Unorm      => ASTC_6x6_LDR, // !macOS
-        //f::Astc6x6Srgb       => ASTC_6x6_sRGB, // !macOS
-        //f::Astc8x5Unorm      => ASTC_8x5_LDR, // !macOS
-        //f::Astc8x5Srgb       => ASTC_8x5_sRGB, // !macOS
-        //f::Astc8x6Unorm      => ASTC_8x6_LDR, // !macOS
-        //f::Astc8x6Srgb       => ASTC_8x6_sRGB, // !macOS
-        //f::Astc8x8Unorm      => ASTC_8x8_LDR, // !macOS
-        //f::Astc8x8Srgb       => ASTC_8x8_sRGB, // !macOS
-        //f::Astc10x5Unorm     => ASTC_10x5_LDR, // !macOS
-        //f::Astc10x5Srgb      => ASTC_10x5_sRGB, // !macOS
-        //f::Astc10x6Unorm     => ASTC_10x6_LDR, // !macOS
-        //f::Astc10x6Srgb      => ASTC_10x6_sRGB, // !macOS
-        //f::Astc10x8Unorm     => ASTC_10x8_LDR, // !macOS
-        //f::Astc10x8Srgb      => ASTC_10x8_sRGB, // !macOS
-        //f::Astc10x10Unorm    => ASTC_10x10_LDR, // !macOS
-        //f::Astc10x10Srgb     => ASTC_10x10_sRGB, // !macOS
-        //f::Astc12x10Unorm    => ASTC_12x10_LDR, // !macOS
-        //f::Astc12x10Srgb     => ASTC_12x10_sRGB, // !macOS
-        //f::Astc12x12Unorm    => ASTC_12x12_LDR, // !macOS
-        //f::Astc12x12Srgb     => ASTC_12x12_sRGB, // !macOS
-        //f::Bgra4Unorm =>
-        //f::R5g6b5Unorm =>
-        //f::A1r5g5b5Unorm =>
-        _ => return None,
-    })
+impl PrivateCapabilities {
+    pub fn map_format(&self, format: Format) -> Option<MTLPixelFormat> {
+        use metal::MTLPixelFormat::*;
+        use hal::format::Format as f;
+        Some(match format {
+            f::B5g6r5Unorm   if self.format_b5 => B5G6R5Unorm,
+            f::B5g5r5a1Unorm if self.format_b5 => BGR5A1Unorm,
+            f::R8Srgb  if self.format_min_srgb_channels <= 1 => R8Unorm_sRGB,
+            f::Rg8Srgb if self.format_min_srgb_channels <= 2 => RG8Unorm_sRGB,
+            f::D24UnormS8Uint if self.format_depth24_stencil8 => Depth24Unorm_Stencil8,
```

Comment:
I was trying to keep the order consistent with `format.rs` because it was a bit hard to spot mismatches